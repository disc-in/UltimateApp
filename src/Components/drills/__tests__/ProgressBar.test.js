import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import ProgressBar from '../ProgressBar';

describe('<ProgressBar />', () => {
  it('renders correctly', async () => {
    const onDotPress = jest.fn();
    const { toJSON, queryAllByTestId } = render(<ProgressBar total={6} current={2} onDotPress={onDotPress} />);

    expect(toJSON()).toMatchSnapshot();

    await fireEvent.press(queryAllByTestId('progress-bar-dot')[3]);
    expect(onDotPress).toBeCalledWith(3);
  });
});
