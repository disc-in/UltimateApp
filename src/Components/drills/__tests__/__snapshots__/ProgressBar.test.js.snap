// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(150, 150, 150)",
          "height": 5,
          "left": 0,
          "position": "absolute",
          "top": 5,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgb(0, 0, 0)",
          "height": 5,
          "left": 0,
          "position": "absolute",
          "top": 5,
          "width": "20%",
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(0, 0, 0)",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 15,
          "opacity": 1,
          "width": 15,
        }
      }
      testID="progress-bar-dot"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(0, 0, 0)",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 15,
          "opacity": 1,
          "width": 15,
        }
      }
      testID="progress-bar-dot"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(150, 150, 150)",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 15,
          "opacity": 1,
          "width": 15,
        }
      }
      testID="progress-bar-dot"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(150, 150, 150)",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 15,
          "opacity": 1,
          "width": 15,
        }
      }
      testID="progress-bar-dot"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(150, 150, 150)",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 15,
          "opacity": 1,
          "width": 15,
        }
      }
      testID="progress-bar-dot"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(150, 150, 150)",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 15,
          "opacity": 1,
          "width": 15,
        }
      }
      testID="progress-bar-dot"
    />
  </View>
</View>
`;
